File: 283_move_zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        if (nums.empty()) return;
        for (int i=0;i<nums.size();i++) {
            if (nums[i]==0) 
                {for (int j=i+1;j<nums.size();j++) {
                    if (nums[j]) {
                        nums[i]=nums[j];
                        nums[j]=0;
                        break;
                        }
                    }
                }
            }
        }
};

File: algorithms/python3/283_move_zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        left=0
        for i in range(len(nums)):
            if nums[i]!=0:
                nums[i],nums[left]=nums[left],nums[i]
                left+=1

a=Solution()

nums=[0,1,0,3,12]

a.moveZeroes(nums)

